src/services/profile/profile-router.test.ts

```diff
@@ -0,0 +1,231 @@
+            points: 0,
+        });
+
+        const response = await getAsAdmin("/profile/id/doesnotexist").expect(404);
+
+        expect(JSON.parse(response.text)).toHaveProperty("error", "UserNotFound");
+    });
+});
+
+describe("DELETE /profile/", () => {
+    it("deletes with a attendee that doesn't exist", async () => {
+        const response = await delAsUser("/profile").expect(404);
+        expect(JSON.parse(response.text)).toHaveProperty("error", "AttendeeNotFound");
+    });
+
+    it("deletes with a valid user", async () => {
+        await Models.AttendeeProfile.create({
+            userId: TESTER.id,
+            displayName: TESTER.name,
+            avatarUrl: TESTER.avatarUrl,
+            discordTag: TESTER.discordTag,
+            points: 0,
+        });
+
+        await Models.AttendeeMetadata.create({
+            userId: TESTER.id,
+            foodWave: 0,
+        });
+
+        const response = await delAsUser("/profile").expect(200);
+        expect(JSON.parse(response.text)).toHaveProperty("success", true);
+    });
+});
+
+describe("GET /profile/leaderboard", () => {
+    it("gets with no limit but only 3 entries", async () => {
+        await Models.AttendeeProfile.create({
+            userId: TESTER.id,
+            displayName: TESTER.name,
+            avatarUrl: TESTER.avatarUrl,
+            discordTag: TESTER.discordTag,
+            points: 25,
+        });
+
+        await Models.AttendeeProfile.create({
+            userId: "tester2",
+            displayName: TESTER.name + "2",
+            avatarUrl: TESTER.avatarUrl,
+            discordTag: TESTER.discordTag,
+            points: 12,
+        });
+
+        await Models.AttendeeProfile.create({
+            userId: "tester3",
+            displayName: TESTER.name + "3",
+            avatarUrl: TESTER.avatarUrl,
+            discordTag: TESTER.discordTag,
+            points: 27,
+        });
+
+        await getAsUser("/profile/leaderboard").expect(200);
+    });
+
+    it("gets with a limit of 2", async () => {
+        await Models.AttendeeProfile.create({
+            userId: TESTER.id,
+            displayName: TESTER.name,
+            avatarUrl: TESTER.avatarUrl,
+            discordTag: TESTER.discordTag,
+            points: 25,
+        });
+
+        await Models.AttendeeProfile.create({
+            userId: "tester2",
+            displayName: TESTER.name + "2",
+            avatarUrl: TESTER.avatarUrl,
+            discordTag: TESTER.discordTag,
+            points: 12,
+        });
+
+        await Models.AttendeeProfile.create({
+            userId: "tester3",
+            displayName: TESTER.name + "3",
+            avatarUrl: TESTER.avatarUrl,
+            discordTag: TESTER.discordTag,
+            points: 27,
+        });
+
+        await getAsUser("/profile/leaderboard?limit=2").expect(200);
+    });
+
+    it("gets with no limit & more entries in db than the max query limit", async () => {
```

Comment:
np: `it("gets with no limit & more entries in db than the max query limit")` -> `it("only gets the max limit when no limit is set")`